File: okreplay-sample/src/main/java/okreplay/sample/Repository.java
package okreplay.sample;

import androidx.annotation.Nullable;

import com.google.auto.value.AutoValue;
import com.squareup.moshi.JsonAdapter;
import com.squareup.moshi.Moshi;

@AutoValue abstract class Repository {
  abstract String name();
  @Nullable abstract String description();

  public static JsonAdapter<Repository> jsonAdapter(Moshi moshi) {
    return new AutoValue_Repository.MoshiJsonAdapter(moshi);
  }
}

File: okreplay-sample/src/main/java/okreplay/sample/Repository.kt
package okreplay.sample

import com.google.auto.value.AutoValue
import com.squareup.moshi.JsonAdapter
import com.squareup.moshi.Moshi
import okreplay.sample.AutoValue_Repository.MoshiJsonAdapter

@AutoValue
internal abstract class Repository {
    abstract fun name(): String?
    abstract fun description(): String?

    companion object {
        @JvmStatic
        fun jsonAdapter(moshi: Moshi?): JsonAdapter<Repository> = MoshiJsonAdapter(moshi)
    }
}
